From 832852cc823334139eb2cd08159e75fcd113fd5a Mon Sep 17 00:00:00 2001
From: MrPowerGamerBR <leonardomalaman@gmail.com>
Date: Sat, 18 Jun 2016 21:22:27 -0300
Subject: [PATCH] [Paper] Be a bit more informative in maxHealth exception


diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftLivingEntity.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftLivingEntity.java
index add554e..e64d5a9 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftLivingEntity.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftLivingEntity.java
@@ -78,7 +78,10 @@ public class CraftLivingEntity extends CraftEntity implements LivingEntity {
 
     public void setHealth(double health) {
         if ((health < 0) || (health > getMaxHealth())) {
-            throw new IllegalArgumentException("Health must be between 0 and " + getMaxHealth());
+        	// Paper - Be more informative
+        	throw new IllegalArgumentException("Health must be between 0 and " + getMaxHealth() + ", but was " + health
+        	        + ". (attribute base value: " + this.getHandle().getAttributeInstance(GenericAttributes.maxHealth).b()
+        	        + (this instanceof CraftPlayer ? ", player: " + ((CraftPlayer) this).getName() + ')' : ')'));
         }
 
         if (entity instanceof EntityPlayer && health == 0) {
-- 
2.7.2.windows.1

